runtime/interpreter/interpreter_statement.go

```diff
@@ -287,6 +287,15 @@ func (interpreter *Interpreter) VisitForStatement(statement *ast.ForStatement) a
 		panic(ExternalError{err})
 	}
 
+	var indexVariable *Variable
+	var one = NewIntValueFromInt64(1)
+	if statement.Index != nil {
+		indexVariable = interpreter.declareVariable(
+			statement.Index.Identifier,
+			NewIntValueFromInt64(0),
+		)
+	}
```

Comment:
Maybe optimize and avoid the allocation when increasing by keeping a plain indexing Go variable and updating the Cadence variable's value in-place:

```suggestion
	var indexVariable *Variable
	var index int64
	var indexValue IntValue
	if statement.Index != nil {
		indexValue = NewIntValueFromInt64(index)
		indexVariable = interpreter.declareVariable(
			statement.Index.Identifier,
			indexValue,
		)
	}
```